### PR TITLE
Return of integrated circuitry + object hearing fix + circuit printer

### PR DESCRIPTION
**Pull request by drexample**
## About The Pull Request

Ports Integrated Circuit Printer - a handy device that contains recipes for all intergrated circuits so you don't have to print them one by one from the lathes.
Returns circuits to RnD and makes them researchable (by adding the printer and an upgrade for it to the designs)
Changes the behaviour of Say code to match what Polaris has - this fixes microphone in implant assembly not working, for example. Things inside of a container inside of a container will hear what's on the outside - what that means is, if you have a voice-activated assembly device implanted or in your backpack, now it will hear what you (or other people) say on the outside. 
Neck smoker implant, anyone?

## Changelog
:cl:
add: Adds circuit printer
fix: Returns circuits to RnD designs
fix: Fixes integrated microphones not working in implants and assemblies not working from containers
/:cl:

### CODE REVIEW

**Review by genessee596 — APPROVED** [review]
Delegate stamp of approval.

Oooh! Integrated circuits! Very nice, glad to see this.

Still needs maintainer approval.

**Review by TorinoFermic — APPROVED** [review]

**Comment by genessee596** [comment]
Approved by maintainers and delegates, merging now.

**Pull request merged**
